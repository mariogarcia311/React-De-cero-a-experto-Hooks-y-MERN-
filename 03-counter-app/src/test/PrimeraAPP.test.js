import PrimeraApp from "../PrimeraAPP"
import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from "enzyme"

describe('pruebas primera APP', () => {
    
    // test('debe mostrar mensaje hola lola', () => {
        
    //     const saludo='hola lola'
    //     const {getByText}=render(<PrimeraApp saludo={saludo} />)

    //     expect(getByText(saludo)).toBeInTheDocument();
    // })
    test('debe de mostrar primeraAPP correctamente ', () => {
        const saludo='hola lola'
        const wrapper= shallow(< PrimeraApp saludo={saludo} />)
        expect(wrapper).toMatchSnapshot();
    })
    test('debe mostrat subtítulo enviado por props', () => {
        
        const saludo='hola lola'
        const subtitulo='soy un subtítulo'
        const wrapper= shallow(
            < PrimeraApp 
                saludo={saludo}
                subtitulo={subtitulo} 
            />
        );
        
        const textoParrafo= wrapper.find('p').text();
        expect(textoParrafo).toBe(subtitulo)
    })
    
    
})
