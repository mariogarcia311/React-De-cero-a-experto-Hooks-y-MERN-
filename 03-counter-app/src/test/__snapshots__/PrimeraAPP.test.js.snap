// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas primera APP debe de mostrar primeraAPP correctamente  1`] = `
<Fragment>
  <h1>
    hola lola
  </h1>
  <p>
    soy un subt√≠tulo
  </p>
</Fragment>
`;
